app/src/main/java/com/example/calculadora/MainActivity.kt

```diff
@@ -36,123 +36,131 @@ class MainActivity : AppCompatActivity() {
         binding.multiplication.setOnClickListener { multiplication() }
         binding.division.setOnClickListener { division() }
         binding.percent.setOnClickListener { percent() }
-        binding.undefined.setOnClickListener { startActivity(Intent(this, Calculate2::class.java))}
+        binding.undefined.setOnClickListener { startActivity(Intent(this, Calculate2::class.java)) }
     }
 
     private fun writeNumber(button: Button) {
-        if (binding.screen.text == "0"){
-            binding.screen.text =""
-        }
-        if (actionTwo == "equal") {
+        if (isDisplayingResult) {
             erase()
-            binding.screen.text = ("${binding.screen.text}${button.text}")
-            actionTwo = ""
-        } else {
-            binding.screen.text = ("${binding.screen.text}${button.text}")
         }
+        binding.screen.text = ("${binding.screen.text}${button.text}")
+
     }
 
     private fun erase() {
         binding.screen.text = ""
-        firstNumber = 0.0
-        secondNumber = 0.0
+        firstNumber = null
+        secondNumber = null
         action = null
-        actionTwo = null
+        isDisplayingResult = false
     }
 
     private fun delete() {
         binding.screen.text = binding.screen.text.dropLast(1)
     }
 
-    private fun savefirstNumberAndErase() {
+    private fun saveFirstNumberAndErase() {
         if (binding.screen.text.toString().isNotBlank()) {
-            firstNumber = binding.screen.text.toString().toDouble()
+            firstNumber = binding.screen.text.toString()
         }
         binding.screen.text = ""
     }
 
     private fun saveSecondNumberAndErase() {
         if (binding.screen.text.toString().isNotBlank()) {
-            secondNumber = binding.screen.text.toString().toDouble()
+            secondNumber = binding.screen.text.toString()
         }
         binding.screen.text = ""
     }
 
-    private fun sum() {
-        if (firstNumber == 0.0) {
-            savefirstNumberAndErase()
+    private fun saveScreen(){
+        if (firstNumber == null) {
+            saveFirstNumberAndErase()
         } else {
             saveSecondNumberAndErase()
         }
+    }
```

Comment:
```suggestion
    private fun saveScreen(){
        if (firstNumber == null) {
            firstNumber = binding.screen.text.toString()
        } else {
            secondNumber = binding.screen.text.toString()
        }
        binding.screen.text = ""
    }
```

And remove functions `saveFirstNumberAndErase` and `saveSecondNumberAndErase`.